org.eclipse.scanning.api/src/org/eclipse/scanning/api/device/DeviceResponse.java

```diff
@@ -96,15 +100,15 @@ public DeviceRequest process(DeviceRequest request) {
 			return error;
 		}
 	}
-	
+
 	private static void processScannables(DeviceRequest request, IScannableDeviceService cservice) throws Exception {

```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 25 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)